test/routes/api/signup.test.js

```diff
@@ -0,0 +1,50 @@
+import { test } from 'node:test';
+import * as assert from 'node:assert';
+import { StatusCodes } from 'http-status-codes';
+
+import { build } from '../../helper.js';
+
+test('/api/signup', async (t) => {
+  const app = await build(t);
+
+  await t.test('POST /signup', async (t) => {
+    await t.test('returns bad request if required fields are missing', async (t) => {
+      const response = await app.inject().post('/api/user/signup').payload({
+      });
+
+      assert.deepStrictEqual(response.statusCode, StatusCodes.BAD_REQUEST);
+    });
+
+    await t.test('returns bad request if passwords do not match', async (t) => {
+      const response = await app.inject().post('/api/user/signup').payload({
+        firstName: 'Jane',
+        lastName: 'Doe',
+        email: 'jane@example.com',
+        password: 'test',
+        confirmPassword: 'differentpassword',
+      });
+
+      assert.deepStrictEqual(response.statusCode, StatusCodes.BAD_REQUEST);
+    });
+
+    await t.test('returns ok with user data on success', async (t) => {
+      const response = await app.inject().post('/api/user/signup').payload({
+        firstName: 'Jane',
+        lastName: 'Doe',
+        email: 'jane@example.com',
+        password: 'test',
+        confirmPassword: 'test',
+      });
+
```

Comment:
One way to debug the error is to print out the response you get from the server. So, right here on line 38, you could add:

```
      console.log(response.body);
```

If you do that, and re-run the code, you should see something in the output like:

```
{"statusCode":500,"error":"Internal Server Error","message":"\nInvalid `prisma.user.create()` invocation:\n\n{\n  data: {\n    firstName: \"Jane\",\n    lastName: \"Smith\",\n    email: \"jane.smith@test.com\",\n+   hashedPassword: String\n  }\n}\n\nArgument `hashedPassword` is missing."}
```

Reply:
The `prisma.user.create` call is failing because you're not passing in the `hashedPassword` which is required.